app/src/main/java/com/mathewsachin/fategrandautomata/accessibility/ScriptRunnerService.kt

```diff
@@ -230,28 +253,40 @@ class ScriptRunnerService : AccessibilityService() {
 
     override fun onInterrupt() {}
 
+    private fun isLandscape() =
+        userInterface.metrics.let { it.widthPixels >= it.heightPixels }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        // Hide overlay in Portrait orientation
+        if (isLandscape()) {
+            Timber.verbose { "LANDSCAPE" }
+
+            if (serviceState is ServiceState.Started) {
+                userInterface.show()
+            }
+        } else {
+            Timber.verbose { "PORTRAIT" }
+
+            userInterface.hide()
+        }
+    }
+
     /**
      * This method is called on any subscribed [AccessibilityEvent] in script_runner_service.xml.
      *
      * When the app in the foreground changes, this method will check if the foreground app is one
-     * of the FGO APKs and will store that information into [currentFgoServer].
+     * of the FGO APKs.
      */
     override fun onAccessibilityEvent(event: AccessibilityEvent?) {
         when (event?.eventType) {
             AccessibilityEvent.TYPE_WINDOW_STATE_CHANGED -> {
                 val foregroundAppName = event.packageName?.toString()
                     ?: return
 
-                val gameServer = GameServerEnum.fromPackageName(foregroundAppName)
+                GameServerEnum.fromPackageName(foregroundAppName)?.let { server ->
```

Comment:
Just for clarity, there should be no case where `fromPackageName` returns null after `onServiceConnected()` was called, right?

But we probably can't replace the ? with a ! because onAccessibilityEvent could be called at the same time as onServiceConnected.

Reply:
We should always avoid `!!`. That defeats the purpose of null-safety.